lib/stdlib/src/maps.erl

```diff
@@ -455,16 +483,38 @@ size(Map) ->
       Map :: #{Key => Value},
       Iterator :: iterator(Key, Value).
 
-iterator(M) when is_map(M) -> [0 | M];
+iterator(M) when is_map(M) -> iterator(M, undefined);
 iterator(M) -> error_with_info({badmap, M}, [M]).
 
+-spec iterator(Map, Order) -> Iterator when
+      Map :: #{Key => Value},
+      Order :: iterator_order(Key),
+      Iterator :: iterator(Key, Value).
+
+iterator(M, undefined) when is_map(M) ->
+    [0 | M];
+iterator(M, ordered) when is_map(M) ->
+    Keys = lists:sort(maps:keys(M)),
```

Comment:
Should we let the sort order be arithmetic or the map-key order which will
- give a strict order where integer and float keys may not get arbitrary ordered
- mimic the legacy order for small maps

Reply:
I'm not sure I understand, what would the arithmetic order look like here versus the term order?

Reply:
I think that we should use the map-key order to ensure a consistent order for numeric keys. I have done that change and added a test.